demos/app/iot_demo.c

```diff
@@ -118,19 +119,63 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
         credentials.pPrivateKey = demoArguments.pPrivateKeyPath;
         credentials.pRootCa = demoArguments.pRootCaPath;
+        credentials.pUserName = NULL;
+        credentials.pPassword = NULL;
+
+        /* Set the MQTT username, as long as it's not empty or NULL. */
+        if( demoArguments.pUserName != NULL )
```

Comment:
Should move these checks to iot_demo_arguments.c (other checks are done there).

Reply:
Done, thank you.